# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/sdn/sdnTextAccessible.h
# Commit: a53fb045a8e3
# Full Hash: a53fb045a8e34c1e1f14ca7b705eb9a5a0e76db2
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-05-02 09:29:52
# Regressor Bug: 1694865
# File Overlap Count: 1
# Description:
#   Bug 1694865 part 19: Make EnumVariant, GeckoCustom and sdn*Accessible classes aggregate MsaaAccessible/MsaaDocAccessible instead of AccessibleWrap/DocAccessibleWrap. r=morgan
#   
#   These objects need to aggregate the primary COM object (MsaaAccessible).
#   Once the IUnknown implementation moves out of AccessibleWrap, it won't be possible to aggregate *AccessibleWrap any more.
#   
# ==============================================================================

diff -r 5bd374b642fe -r a53fb045a8e3 accessible/windows/sdn/sdnTextAccessible.h
--- a/accessible/windows/sdn/sdnTextAccessible.h	Sat May 01 21:09:04 2021 +0000
+++ b/accessible/windows/sdn/sdnTextAccessible.h	Sat May 01 22:29:25 2021 +0000
@@ -10,7 +10,7 @@
 #include "ISimpleDOM.h"
 #include "IUnknownImpl.h"
 
-#include "AccessibleWrap.h"
+#include "MsaaAccessible.h"
 
 class nsIFrame;
 struct nsPoint;
@@ -20,8 +20,7 @@
 
 class sdnTextAccessible final : public ISimpleDOMText {
  public:
-  explicit sdnTextAccessible(AccessibleWrap* aAccessible)
-      : mAccessible(aAccessible){};
+  explicit sdnTextAccessible(MsaaAccessible* aMsaa) : mMsaa(aMsaa){};
   ~sdnTextAccessible() {}
 
   DECL_IUNKNOWN
@@ -61,7 +60,7 @@
   nsIFrame* GetPointFromOffset(nsIFrame* aContainingFrame, int32_t aOffset,
                                bool aPreferNext, nsPoint& aOutPoint);
 
-  RefPtr<AccessibleWrap> mAccessible;
+  RefPtr<MsaaAccessible> mMsaa;
 };
 
 }  // namespace a11y
